static/src/js/containers/MapContainer/MapEvents.jsx

```diff
@@ -23,6 +62,19 @@ const MapEvents = (props) => {
       attributionElement.innerHTML = '* © <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
       layersControl.appendChild(attributionElement)
     }
+
+    // Return cleanup function to remove event listeners when component unmounts
+    return () => {
```

Comment:
This is a small thing, and since this is going to be short-lived Im not sure its even worth fixing, but It would be better to do your looping and finding in the scope above and saving it to a variable that could be used in the return function, rather than duplicating the logic down here.